Core/include/Acts/Utilities/Helpers.hpp

```diff
@@ -192,92 +192,6 @@ inline ActsMatrixD<3, 3> cross(const ActsMatrixD<3, 3>& m, const Vector3D& v) {
   return r;
 }
 
-/// @brief Access to the time component of input parameter
-///
-/// @param spacePointVec The SpacePointVector
-/// @return Reference to the time component
-inline ParValue_t& time(SpacePointVector& spacePointVec) {
```

Comment:
I totally agree with Fabian here. I would prefer keeping these helper functions in order to make the code more readable.

Reply:
I would really prefer to remove this. This adds two different ways to access the same thing. Since we must use the index-based access anyways for the matrices, we should stick to that way everywhere. I also think that assignment to the return value of a function call is unintuitive, i.e. I strongly prefer
```
params[eBoundTime] = ...;
```
over
```
time(params) = ...;
```

There is an additional problem, that the helper methods are only defined for a few particular types. They do not support e.g. Eigen maps or views into larger matrices, and I expect that we will add more such data storage in the near future. Using indices there is universal, while for the helper functions we would have to add them for all specific types.

Reply:
For a single component I agree with you but the `segment`, `head`, `tail` methods are less intuitive and so is the getter for position, momentum etc.

Reply:
To get this PR in quickly, I propose that we keep the replacement of the `time(...)` helper with regular index-based access but leave the `position(...)` helper untouched. We can discuss potential removal in a separate issue and later PR.